extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -503,23 +508,44 @@ public String getHlsUrl() throws ParsingException {
     @Override
     public List<AudioStream> getAudioStreams() throws ExtractionException {
         assertPageFetched();
-        final List<AudioStream> audioStreams = new ArrayList<>();
-        final YoutubeThrottlingDecrypter throttlingDecrypter = new YoutubeThrottlingDecrypter(getId());
 
-        try {
-            for (final Map.Entry<String, ItagItem> entry : getItags(ADAPTIVE_FORMATS,
-                    ItagItem.ItagType.AUDIO).entrySet()) {
-                final ItagItem itag = entry.getValue();
-                String url = entry.getKey();
-                url = throttlingDecrypter.apply(url);
-
-                final AudioStream audioStream = new AudioStream(url, itag);
-                if (!Stream.containSimilarStream(audioStream, audioStreams)) {
-                    audioStreams.add(audioStream);
+        if (audioStreams == null) {
+            audioStreams = new ArrayList<>();
+
+            try {
+                for (final ItagInfo itagInfo : getItags(ADAPTIVE_FORMATS,
+                        ItagItem.ItagType.AUDIO)) {
+                    final ItagItem itag = itagInfo.getItagItem();
+                    final String content = itagInfo.getContent();
+                    final boolean isUrl = itagInfo.getIsUrl();
```

Comment:
I think you can remove this local variable and just inline it in the `if` below (not for the others though, as they're used multiple times)
```suggestion
```